aspnet/fundamentals/distributed-cache.rst

```diff
@@ -1,10 +1,106 @@
+-------------------------------
+The ``IDistributedCache`` interface includes both synchronous and async methods. The interface allows items to get added, retrieved, and removed from the distributed cache implementation. The complete list of methods supported by the interface is:
+
+Connect, ConnectAsync
+	*Deprecated*
+
+Get, GetAsync
+	Takes a string key and retrieves a cached item as a ``byte[]`` if found in the cache.
+	
+Set, SetAsync
+	Adds an item (as ``byte[]``) to the cache using a string key.
+	
+Refresh, RefreshAsync
+	Refreshes an item in the cache based on its key, resetting its sliding expiration timeout (if any).
+	
+Remove, RemoveAsync
+	Removes a cache entry based on its key.
+
+In your ASP.NET 5 application, you can configure the specific implementation of ``IDistributedCache`` in your ``Startup`` class, and add it to the container in the ``ConfigureServices`` method. Of course, you will also need to specify any necessary dependencies in ``project.json``. Once this configuration is in place, your application's :doc:`middleware` or MVC Controller classes can request an instance of ``IDistributedCache`` via their constructor, and these will be provided via :doc:`dependency-injection` (DI).
+
+.. note:: There is no need to use a Singleton or Scoped lifetime for ``IDistributedCache`` instances (at least for the built-in implementations). You can also create an instance wherever you might need one (instead of using :doc:`dependency-injection`), but this can make your code harder to test, and violates the `Explicit Dependencies Principle <http://deviq.com/explicit-dependencies-principle/>`_.
+
+The following code example shows how to use an instance of ``IDistributedCache`` in one of your ASP.NET 5 application's classes (in this case, a simple middleware):
+
+.. literalinclude:: dist-cache/sample/src/DistCacheSample/StartTimeHeader.cs
+  :language: c#
+  :linenos:
+  :emphasize-lines: 13,16,19,25-29
+
+In the above example, the value is read from, but never written to, the cache. This is because in this case the value is only written when the server starts up. Since getting and setting values from the cache must use a ``byte[]``, the string value being stored must be converted using ``Encoding.UTF8.GetString``. Similarly, the string value is converted into a ``byte[]`` (in this case using ``Encoding.UTF8.GetBytes``) when it is set during application startup. The following code from ``Startup.cs`` shows the value being set:
+
+.. literalinclude:: dist-cache/sample/src/DistCacheSample/Startup.cs
+  :language: c#
+  :linenos:
+  :lines: 59-68
+  :emphasize-lines: 6,9
+
+.. note:: Normally you would request an instance of ``IDistributedCache`` via DI, rather than fetching it from ``ApplicationServices`` (as shown above on line 8). However, during application startup only certain types are available via DI, so this service locator pattern approach when access to cache instances is needed within the ``Configure`` method.
```

Comment:
You should be able to inject IDistributedCache (and other services from ConfigureServices) into the Configure method as parameters.